sync-core/src/main/java/com/cloudant/sync/datastore/BasicDatastore.java

```diff
@@ -1968,7 +2034,7 @@ public BasicDocumentRevision call(SQLDatabase db) throws Exception {
         } catch (InterruptedException e) {
             logger.log(Level.SEVERE,"Failed to delete document",e);
         } catch (ExecutionException e) {
-            logger.log(Level.SEVERE,"Failed to delete document",e);
+            throw new DocumentException("Failed to delete document",e);
```

Comment:
Java's standard stack traces normally look like 

```
some.exception.here: a message here
some.class.name(name.java:80)
....
caused by caused by message here
some.other.class.name(name.java 99)
```

so I think we can get away with not appending the error message